src/scikit_build_core/program_search.py

```diff
@@ -77,7 +77,9 @@ def get_cmake_programs(*, module: bool = True) -> Generator[Program, None, None]
             continue
 
         try:
-            version = Version(result.stdout.splitlines()[0].split()[-1])
+            version = Version(
+                result.stdout.splitlines()[0].split()[-1].split("-msvc")[0]
```

Comment:
```suggestion
                result.stdout.splitlines()[0].split()[-1].split("-")[0]
```

I think it's fine to always remove any extension here. These are usually used to indicate extra features or a build number, which shouldn't cause it to be invalid.